### DISCUSSION TITLE
Meeting notes

### DISCUSSION THREAD

**Discussion by keunes** [OpenPodcastAPI/api-specs#35] [Announcements]
For every meeting we have, we aim to draw up some quick notes to remind ourselves of what we discussed and agreed upon. To not loose track and be transparent, we'll try and list here all the meeting notes we have.

**Comment by keunes** [comment]
2022-10-06: https://pad.funkwhale.audio/kKu5TOhcQYG5it8z6C9gbQ#

**Reply by JonOfUs** [reply]
Topics: problems with gpodder, expectations, participating projects

**Comment by keunes** [comment]
2023-03-14: https://pad.funkwhale.audio/ilMy818PSMOPQGonUf6N_Q

**Reply by JonOfUs** [reply]
Topics: subscriptions GUID, subscriptions deletion

**Comment by keunes** [comment]
2023-04-11: https://pad.funkwhale.audio/06RybkKBQ3Wirr2by_ULWA

**Reply by JonOfUs** [reply]
Topics: subscriptions fetch/delete, repository

**Comment by georgkrause** [comment]
2023-05-30: https://pad.funkwhale.audio/oCfs5kJ6QTu02d_oVHW7DA

**Reply by JonOfUs** [reply]
Topics: episodes endpoints, episodes data fields 1, episodes identification 1

**Comment by JonOfUs** [comment]
2023-07-11: https://pad.funkwhale.audio/FkuIqtPGT-ynYKqBieHffw

**Reply by JonOfUs** [reply]
Topics: episodes data fields 2, episode identification 2

**Comment by keunes** [comment]
2023-08-01: https://pad.funkwhale.audio/CkkD0nY_TniUmTQgHx0e_w

**Reply by JonOfUs** [reply]
Topics: versioning, versioning endpoints 1, capabilities endpoints

**Reply by georgkrause** [reply]
@keunes I might be pedantic here, but the meeting was on 2023-08-01

**Comment by JonOfUs** [comment]
2023-09-05: https://pad.funkwhale.audio/s/T-yx14DsH#

**Reply by JonOfUs** [reply]
Topics: Starlight migration, versioning endpoints 2, episodes identification 3 (final?)

**Comment by JonOfUs** [comment]
2023-10-03: https://pad.funkwhale.audio/s/uzRUuWzCW#

**Reply by JonOfUs** [reply]
Topics: NLnet funding, Netlify/Starlight, episodes endpoints, subscriptions implementation

**Comment by keunes** [comment]
2024-02-27: https://pad.funkwhale.audio/s/88C5eXrRq

**Reply by JonOfUs** [reply]
Topics: episodes data fields 3, next meetings

**Comment by JonOfUs** [comment]
2024-04-16: https://pad.funkwhale.audio/s/6mWuDexgz

**Reply by keunes** [reply]
Topics: handing duplicate subscription, episode & subscription data timestamps, episode guids (briefly)

**Comment by JonOfUs** [comment]
2024-04-24: https://pad.funkwhale.audio/kIRwEOYDRNqTA4np6vbBVg#

**Reply by keunes** [reply]
Topics: episode identification (3 new test cases), smart server design (avoiding data duplication), lazy synchronization (informing server of new episodes only in case of additional user data)

**Comment by JonOfUs** [comment]
2024-05-06: https://pad.funkwhale.audio/2iNOQ2N2RF-ICMQcPWyyYQ#

**Reply by JonOfUs** [reply]
Topics: episode identification, fallback episode matching, episode endpoint roadmap

**Comment by JonOfUs** [comment]
2024-05-28: https://pad.funkwhale.audio/bBypcPwdTCydvmoOLsM4ng

**Reply by JonOfUs** [reply]
Topics: episode endpoints assembly, episode data fields, discussion on pagination and max batch size

**Comment by JonOfUs** [comment]
2024-06-19: https://pad.funkwhale.audio/PaVqiT8rR1y31gmRKjs4Pg

**Reply by JonOfUs** [reply]
Topics: episode endpoints assembly, episode data fields

**Comment by keunes** [comment]
2024-11-07: https://pad.funkwhale.audio/s/0788dnUKe

**Comment by keunes** [comment]
2024-11-21: https://pad.funkwhale.audio/Ms-niTP9S0a9mMXXD7n--g#

**Reply by keunes** [reply]
Topics: fundamental question about the approach, versioning- & capabilities

**Reply by georgkrause** [reply]
I was just reading through the protocol and now I am wondering if the following JSON is really what we agreed upon?

```json
{
  "capabilities": {
     "openpodcastapi::betav1",
     "openpodcastapi::core",
     "openpodcastapi::corev2",
     "openpodcastapi::optional": [
        "coversync",
        "queuesync"
     ],
     "openpodcastapi::optional::v2": [
        "coversync",
        "queuesync"
     ],
     "othergroup::socialsharing"
  }
}
```

Didn't we decided that having a single array with identifiers that contain the version would be the better solution? @mogwa1 @keunes

**Reply by keunes** [reply]
Actually I think I directly copied that from the previous meeting, without changes here.

> identifiers that contain the version would be the better solution?

My understanding was that that was true for _core_. For optional endpoints we had discussions on whether they should align with core versioning and that it shouldn't, but otherwise I don't recall conclusions there.

And whether we should have the optional endpoints have the version be part of the name (like for core) I'm not sure, as I would expect there's more versions than for core. While at it: how would we do betas for optional endpoints?
